hibernate-core/src/main/java/org/hibernate/loader/ast/internal/AbstractNaturalIdLoader.java

```diff
@@ -141,18 +144,33 @@ public Callback getCallback() {
 			);
 		}
 
-		return results.get( 0 );
+		final T result = results.get( 0 );
+		postLoadListener.completedLoadByNaturalId( entityDescriptor, result, session );
+
+		return result;
 	}
 
+	protected abstract Object resolveNaturalIdBindValue(Object naturalIdToLoad, SharedSessionContractImplementor session);
+
+	protected abstract void applyNaturalIdAsJdbcParameters(
+			Object naturalIdToLoad,
+			List<JdbcParameter> jdbcParameters,
```

Comment:
In theory `List<? extends JdbcParameter>` is more flexible, unless we are sure JdbcParameter could never have child class.